pkg/controller/apiclient/apiclient_util_test.go

```diff
@@ -14,18 +14,180 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package apiclient
+package apiclient_test
 
 import (
+	"fmt"
 	"testing"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	. "github.com/kubernetes-incubator/service-catalog/pkg/controller/apiclient"
+	"github.com/kubernetes-incubator/service-catalog/pkg/controller/apiclient/mem"
+	kapi "k8s.io/kubernetes/pkg/api"
 )
 
-// TODO: implement these (https://github.com/kubernetes-incubator/service-catalog/issues/134)
-func TestGetServicePlanInfo(t *testing.T) {
+func TestGetServicePlanInfoEmpty(t *testing.T) {
+	apiclient := mem.NewAPIClient()
+	if _, _, _, err := GetServicePlanInfo(apiclient.ServiceClasses(), "service", ""); err == nil {
+		t.Error("Expected err on GetServicePlanInfo with no plan with empty apiclient")
+	}
+	if _, _, _, err := GetServicePlanInfo(apiclient.ServiceClasses(), "service", "plan"); err == nil {
+		t.Error("Expected err on GetServicePlanInfo with plan with empty apiclient")
+	}
+}
+
+// test service classes with single plan
+func TestGetServicePlanInfoOne(t *testing.T) {
+	apiclient := mem.NewAPIClient()
+	svcClss := createServiceClasses(2, 1)
+	apiclient.ServiceClasses().Create(&svcClss[0])
+
+	// test that getting with empty plan name works
+	if err := testGetServicePlanInfo(apiclient, &svcClss[0], &svcClss[0].Plans[0], ""); err != nil {
+		t.Error(err)
+	}
+
+	// test that getting with correct plan name works
+	if err := testGetServicePlanInfo(apiclient, &svcClss[0], &svcClss[0].Plans[0],
+		svcClss[0].Plans[0].Name); err != nil {
+
+		t.Error(err)
+	}
+
+	// test that using wrong plan name is wrong
+	if err := testGetServicePlanInfo(apiclient, &svcClss[0], &svcClss[0].Plans[0],
+		"wrong_name"); err == nil {
+
+		t.Error("Expected err when using wrong plan name")
+	}
+
+	// test that using wrong service name is wrong
+	if err := testGetServicePlanInfo(apiclient, &svcClss[1], &svcClss[0].Plans[0],
+		svcClss[0].Plans[0].Name); err == nil {
+
+		t.Error("Expected err when using wrong service name")
+	}
+}
+
+// test service classes with multiple plans
+func TestGetServicePlanInfoMany(t *testing.T) {
+	apiclient := mem.NewAPIClient()
+	svcClss := createServiceClasses(3, 3)
```

Comment:
@pmorie @MHBauer this creates 3 services classes each with 3 service plans.

I would agree it's not obvious at first what it does, but I'm also not too sure how to improve it. I can add a comment above the function definition, but perhaps you have a better idea for naming or something?